Sources/_MatchingEngine/Regex/AST/Atom.swift

```diff
@@ -443,6 +449,53 @@ extension AST.Atom {
   }
 }
 
+extension AST.Atom {
+  public struct Callout: Hashable {
+    public enum Argument: Hashable {
+      case number(Int)
+      case string(String)
+    }
+    public var arg: AST.Located<Argument>
+    public init(_ arg: AST.Located<Argument>) {
+      self.arg = arg
+    }
+  }
+}
+
+extension AST.Atom {
+  public struct BacktrackingDirective: Hashable {
+    public enum Kind: Hashable {
+      /// (*ACCEPT)
+      case accept
+      /// (*FAIL*)
+      case fail
+      /// (*MARK:NAME)
+      case mark
+      /// (*COMMIT)
+      case commit
+      /// (*PRUNE)
+      case prune
+      /// (*SKIP)
+      case skip
+      /// (*THEN)
+      case then
+    }
+    public var kind: AST.Located<Kind>
+    public var name: AST.Located<String>?
```

Comment:
Is name orthogonal to kind, or is `.mark` the only one that can have a name?

Reply:
They all can optionally have names, but `.mark` must have a name